cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1277,6 +1285,25 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
                                 }
 
                                 this.kafkaExternalBootstrapDnsName.add(bootstrapAddress);
+
+                                Kafka resource = crdOperator.get(namespace, name);
+                                KafkaStatus ks = resource.getStatus();
+                                ListenersStatus ls = new ListenersStatus();
+
+                                if (ks == null) {
+                                    ks = new KafkaStatus();
+                                }
+
+                                if (ks.getListeners() != null) {
+                                    ls.setPlain(ks.getListeners().getPlain());
+                                    ls.setTls(ks.getListeners().getTls());
+                                }
+                                String externalBootstrapHost = serviceName + "." + namespace + ".svc";
+                                ls.setExternal(new ListenerStatus(externalBootstrapHost, kafkaCluster.getExternalPort(), null));
```

Comment:
This seems to set it correctly for load balancers. but not for node port external listener. For that you might need to set it as well ... with the address being the node address and the port being the node port.

Reply:
fixed